src/main/scala/akka/contrib/process/BlockingProcess.scala

```diff
@@ -107,16 +110,24 @@ class BlockingProcess(
     }
 
     try {
-      val stdin =
-        context.actorOf(OutputStreamSubscriber.props(process.getOutputStream), "stdin")
-
-      val stdout =
-        context.watch(context.actorOf(InputStreamPublisher.props(process.getInputStream, stdioTimeout), "stdout"))
-
-      val stderr =
-        context.watch(context.actorOf(InputStreamPublisher.props(process.getErrorStream, stdioTimeout), "stderr"))
-
-      context.parent ! Started(ActorSubscriber(stdin), ActorPublisher(stdout), ActorPublisher(stderr))
+      val stdin = Sink.outputStream(process.getOutputStream)
+        .mapMaterializedValue { futureResult =>
+          futureResult.onComplete(_ => context.stop(self))
+          futureResult
+        }
+
+      val stdout = Source.inputStream(process.getInputStream).withAttributes(Attributes.inputBuffer(1, 1))
+        .mapMaterializedValue { futureResult =>
+          futureResult.onComplete(_ => context.stop(self))
+          futureResult
+        }
+
+      val stderr = Source.inputStream(process.getErrorStream).withAttributes(Attributes.inputBuffer(1, 1)).mapMaterializedValue { futureResult =>
+        futureResult.onComplete(_ => context.stop(self))
+        futureResult
+      }
```

Comment:
@2m - I've set the Source attributes as creation time like you suggested (as opposed to materialization time):

https://github.com/2m/akka-contrib-extra/commit/35faea520dca7d21abcc03b1c3598c7416039547